packages/teleport/src/Support/ScheduleUpgrades/ScheduleUpgrades.tsx

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { ButtonPrimary, ButtonSecondary } from 'design/Button';
+import Dialog, {
+  DialogContent,
+  DialogFooter,
+  DialogTitle,
+} from 'design/Dialog';
+import Select, { Option } from 'shared/components/Select';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import { UpgradeWindowStart } from 'teleport/services/upgradeWindow/upgradeWindow';
+import Text from 'design/Text';
+import Alert from 'design/Alert';
+
+export function ScheduleUpgrades({
+  onClose,
+  onSave,
+  upgradeWindowOptions,
+  selectedWindow,
+  onSelectedWindowChange,
+  attempt,
+}: Props) {
+  const options =
+    attempt.status === 'processing'
+      ? []
+      : upgradeWindowOptions.map((window: UpgradeWindowStart) => ({
+          label: makeLabel(window),
+          value: window,
+        }));
```

Comment:
```suggestion
  let options = []
  if (attempt.status === 'success') {
    options = upgradeWindowOptions.map((window: UpgradeWindowStart) => ({
          label: makeLabel(window),
          value: window,
        }));
  }
```
easier to read imo. also i think we are over complicating this. the update window times are just consts, which is nice, then we can just move this outside of the component:

```ts
const upgradeWindowOptions = [...]

...
   const options = attempt.status === 'processing' ? [] : upgradeWindowOptions

```

Reply:
Yes, this is easier to read, thanks!

I'd like to keep the options being a result of `.map()` so we have a distinction between the option and the actual value (easier to track since the value itself will be a state in the hook).

What doo you think about doing something like this?

```jsx
const availableUpgradeWindowStarts: UpgradeWindowStart[] = [
  '08:00:00',
  '16:00:00',
  '23:00:00',
];

const upgradeWindowOptions = availableUpgradeWindowStarts.map(
  (window: UpgradeWindowStart) => ({
    label: makeLabel(window),
    value: window,
  })
);

export function ScheduleUpgrades({
  const options = attempt.status === 'processing' ? [] : upgradeWindowOptions
// ...
```

I implemented this in https://github.com/gravitational/webapps/pull/980/commits/58e4996d1bee3d941bce01650930183f354e08e2, please let me know what you think